scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+package com.bittercode.service.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.constant.db.BooksDBConstants;
+import com.bittercode.model.Book;
+import com.bittercode.model.StoreException;
+import com.bittercode.service.BookService;
+import com.bittercode.util.DBUtil;
+
+public class BookServiceImpl implements BookService {
+
+    private static final String getAllBooksQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK;
+    private static final String getBookByIdQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+            + " WHERE " + BooksDBConstants.COLUMN_BARCODE + " = ?";
+
+    private static final String deleteBookByIdQuery = "DELETE FROM " + BooksDBConstants.TABLE_BOOK + "  WHERE "
+            + BooksDBConstants.COLUMN_BARCODE + "=?";
+
+    private static final String addBookQuery = "INSERT INTO " + BooksDBConstants.TABLE_BOOK + "  VALUES(?,?,?,?,?)";
+
+    private static final String updateBookQtyByIdQuery = "UPDATE " + BooksDBConstants.TABLE_BOOK + " SET "
+            + BooksDBConstants.COLUMN_QUANTITY + "=? WHERE " + BooksDBConstants.COLUMN_BARCODE
+            + "=?";
+
+    private static final String updateBookByIdQuery = "UPDATE " + BooksDBConstants.TABLE_BOOK + " SET "
+            + BooksDBConstants.COLUMN_NAME + "=? , "
+            + BooksDBConstants.COLUMN_AUTHOR + "=?, "
+            + BooksDBConstants.COLUMN_PRICE + "=?, "
+            + BooksDBConstants.COLUMN_QUANTITY + "=? "
+            + "  WHERE " + BooksDBConstants.COLUMN_BARCODE
+            + "=?";
+
+    @Override
+    public Book getBookById(String bookId) throws StoreException {
+        Book book = null;
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(getBookByIdQuery);
+            ps.setString(1, bookId);
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                String bCode = rs.getString(1);
+                String bName = rs.getString(2);
+                String bAuthor = rs.getString(3);
+                int bPrice = rs.getInt(4);
+                int bQty = rs.getInt(5);
+
+                book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+            }
+        } catch (SQLException e) {
+
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> getAllBooks() throws StoreException {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The getAllBooks() method retrieves all books from the database without any pagination or limit, which could lead to performance issues and excessive memory usage when dealing with a large number of books.
</div>
<div id="fix">
<b>Fix:</b> Implement pagination or limit the number of books retrieved in a single query to improve scalability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -64,9 +64,11 @@
  @Override
 -    public List<Book> getAllBooks() throws StoreException {
 +    public List<Book> getAllBooks(int page, int pageSize) throws StoreException {
          List<Book> books = new ArrayList<Book>();
          Connection con = DBUtil.getConnection();
 +        int offset = (page - 1) * pageSize;
 +        String paginatedQuery = getAllBooksQuery + " LIMIT ? OFFSET ?";
         
          try {
 -            PreparedStatement ps = con.prepareStatement(getAllBooksQuery);
 +            PreparedStatement ps = con.prepareStatement(paginatedQuery);
 +            ps.setInt(1, pageSize);
 +            ps.setInt(2, offset);
              ResultSet rs = ps.executeQuery();
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged